src/app/core/csp.service.ts

```diff
@@ -0,0 +1,27 @@
+import { DOCUMENT } from '@angular/common';
+import { Inject, Injectable } from '@angular/core';
+import { environment } from '../../environments/environment';
+
+
+@Injectable()
+export class CspService {
+  constructor(
+    @Inject(DOCUMENT) private doc: Document
+  ) { }
+
+  register() {
+    if (environment.name !== 'dev') {
+      const meta: HTMLMetaElement = this.doc.createElement('meta');
+      meta.setAttribute('http-equiv', 'Content-Security-Policy');
+      meta.setAttribute(
+        'content',
+        `default-src 'self';
+         child-src 'none';
+         font-src 'self' data:;
+         style-src 'unsafe-inline';
+         script-src 'self' 'unsafe-eval'`
```

Comment:
Anyway to not use unsafe settings? If not, I'm fine merging this now, but perhaps we need a new issue to change the code so these are not needed?

Reply:
Good job pushing me to do better. I was able to get rid of 'unsafe-eval' and beefed up CSP with couple more directives. Have to keep 'unsafe-inline' due to https://github.com/angular/angular/issues/6361.